---
title: "Computational Reproducibility"
author: "Fred LaPolla"
date: "10/14/2020"
output: html_document
---

```{r setup, include=TRUE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown file

This file, an R Markdown lets you combine plain text, in the white spaces, and code chunks, in the gray. The text is not really code, but can be used to communicate, while the code chunks let you combine the actual analysis you are running alongside the results. This means someone can see exactly what you did to get your results.

To add a chunk either click on the green Insert button and choose R or hit Command + Option + i (Mac) or Ctrl + Alt + i (PC)

```{r}
3+3
```

## Homework

1) Create an R Project to pull this file from GitHub. 
Done by Alan Varghese, name is Alanvarg

2) Please describe one of the ways that Wilson et. al. suggest is a good practice for computational reproducibility and how a tool like R can be used. Write your answer below.

Writing code that is easy to read/understand for every interested party
______________
One of my main fields of interests involves DNA and RNA sequencing, and hence there is a significant chunk of computational data analysis involved. While there are very good packages presently to analyze RNA-Seq or other very popular sequencing datasets such as ChIP-seq, these days newer ways to look at existing data, as well as new types of sequencing methods are coming out every week.

This necessitates readers/researchers like me to be able to easily read the codes used in the paper. The same also applies when I am doing my own data analysis to ensure my collaborators and future self can do the same.

The good practices mentioned in Wilson et al., with respect to software such as, 
i)   Adding useful/helpful comments for each function/line(s) of code ii)  Adding a simple example data set to test/implement the code
iii) Mentioning all background packages/dependencies required

etc., are very valuable suggestions/recommendations that will go a long way in terms of making new sequencing data analysis easier and faster to understand and implement as well as making it reproducible, which will definitely speed up research in the field. 

R, especially through RStudio would definitely help in this process, as RStudio enables easy commenting/readability, extraction of datasets and most importantly is widely used in the community, enabling faster dissemination of newer data analysis methods and also their reproducibility.

3) Add a code chunk and find the square root of 4 to the 13th power. If you are unsure of what command to use, google it. 
```{r}
#Below is the code to first find 4 rasied to its 13th power and then its square root 
sqrt(4^13)
```


4) Push this homework to the GitHub repository for the homework. 

Possibly due to me using a Windows system, I cannot commit my changes to the GitHub repository. I have hence uploaded this assignment (as a pdf) to Brightspace and also uploaded this rmd file.I will try to solve this issue as soon as possible and then upload it to GitHub later.

git config --global user.email "alan.varghese@nyulangone.org"
git config --global user.name "Alan Varghese"

```{r}


```


